### PR TITLE
update Kubernetes programming guide to spark2.3

### PR DESCRIPTION
**Pull request by megaSpoon**
## What changes were proposed in this pull request?

* Kubernetes guide update to be compatible with Spark 2.3
* ships a dockerfile that can be deployed into containers within pods with Spark2.3 and latest BigDL

## How was this patch tested?

run the scala and python example in the "Run On Kubernetes" programming guide within the new docker image 

## Related links or issues (optional)
https://bigdl-project.github.io/0.6.0/#ProgrammingGuide/run-on-k8s/
https://spark.apache.org/docs/2.3.0/running-on-kubernetes.html#docker-images

### CODE REVIEW

**Comment by megaSpoon** [comment]
Working on the new Dockerfile now

File: docs/docs/ProgrammingGuide/run-on-k8s.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ $SPARK_HOME/bin/spark-submit \
   --conf spark.kubernetes.executor.docker.image=intelanalytics/spark-executor:v2.2.0-kubernetes-0.5.0-ubuntu-14.04 \
   --conf spark.kubernetes.initcontainer.docker.image=intelanalytics/spark-init:v2.2.0-kubernetes-0.5.0-ubuntu-14.04 \
   --conf spark.kubernetes.resourceStagingServer.uri=http://<address-of-any-cluster-node>:31000 \
-  $BIGDL_HOME/lib/bigdl-0.4.0-SNAPSHOT-jar-with-dependencies.jar \
+  $BIGDL_HOME/lib/bigdl-0.6.0-SNAPSHOT-jar-with-dependencies.jar \
```
Comment (yiheng) [inline docs/docs/ProgrammingGuide/run-on-k8s.md]: why 0.6.0-SNAPSHOT? The code version is 0.7.0-SNAPSHOT, so should it be 0.7.0-SNAPSHOT? And when release it should be changed to 0.7.0, right?

File: docs/docs/ProgrammingGuide/run-on-k8s.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-2. You need to spin up the **resource staging server** for dependency management. See [Dependency Management](https://apache-spark-on-k8s.github.io/userdocs/running-on-kubernetes.html#dependency-management) (This is optional if all your application dependencies are
-packaged into your own custom docker image or resides in remote locations like HDFS. See [Dependency Management Without The Resource Staging Server](https://apache-spark-on-k8s.github.io/userdocs/running-on-kubernetes.html#dependency-management-without-the-resource-staging-server))
+You need to have a running Kubernetes cluster that support Spark on Kubernetes. See [here](https://spark.apache.org/docs/2.3.0/running-on-kubernetes.html#prerequisites)
 
 ---
 ## **Docker images**
+For Spark2.3,
+BigDL ships with a Dockerfile that can be found in the ```kubernetes/dockerfiles/``` directory.
 
+For Spark2.2,
```
Comment (yiheng) [inline docs/docs/ProgrammingGuide/run-on-k8s.md]: Remove Spark 2.2? As the k8s is officially supported in Spark 2.3.

Comment (megaSpoon) [reply docs/docs/ProgrammingGuide/run-on-k8s.md]: Ok.

File: docs/docs/ProgrammingGuide/run-on-k8s.md (hunk 1 above)
Comment (megaSpoon) [reply docs/docs/ProgrammingGuide/run-on-k8s.md]: exactly! The part will be changed.

**Comment by yiheng** [comment]
@megaSpoon any update?

**Comment by megaSpoon** [comment]
tested with dockerfile, create the image and run on the minikube. It works with Lenet5 example but I uploaded the mnist manually to the image which is not elegant.

Test with hdfs next.

File: docker/BigDL/bigdl-k8s/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+WORKDIR /opt/work
+
+ARG BIGDL_VERSION=0.6.0
+ARG SPARK_VERSION=2.3.1
+ENV BIGDL_VERSION_ENV		${BIGDL_VERSION}
+ENV SPARK_VERSION_ENV		${SPARK_VERSION}
+ENV SPARK_HOME			/opt/work/spark-${SPARK_VERSION}
+ENV BIGDL_HOME			/opt/work/bigdl-${BIGDL_VERSION}
+ENV BIGDL_TUTORIALS_HOME	/opt/work/BigDL-Tutorials
```
Comment (yiheng) [inline docker/BigDL/bigdl-k8s/Dockerfile]: we should not put BIGDL_TUTORIALS here, right?

Comment (yiheng) [reply docker/BigDL/bigdl-k8s/Dockerfile]: we should keep the installed package minimal.

**Comment by yiheng** [comment]
I think you need to modify from the docker files under: https://github.com/intel-analytics/BigDL/tree/master/docker/spark-on-k8s

**Comment by megaSpoon** [comment]
re-modify the file under https://github.com/intel-analytics/BigDL/tree/master/docker/spark-on-k8s. I see from Spark 2.3 on kubernetes document, that it gets rid of specifying different images for driver, executors. It looks like it was dealt behind the scenes behind the entrypoint.sh

**Comment by megaSpoon** [comment]
So basically I combined these dockerfiles into one and removed the redundant part.

File: docs/docs/ProgrammingGuide/run-on-k8s.md [hunk 4]
```diff
@@ -1,82 +1,129 @@
-The [Apache Spark on Kubernetes](https://apache-spark-on-k8s.github.io/userdocs/index.html) project enables
-native support for submitting Spark application to a kubernetes cluster. As a deep learning library for Apache
+Spark can run on clusters managed by Kubernetes. This feature makes use of native Kubernetes scheduler that has been added to Spark 2.3.now As a deep learning library for Apache
```
Comment (yiheng) [inline docs/docs/ProgrammingGuide/run-on-k8s.md]: .now As?

**Comment by yiheng** [comment]
can we publish the prebuild docker image on our docker hub?

**Comment by megaSpoon** [comment]
>           can we publish the prebuild docker image on our docker hub?

Sure then our docker hub username and password is needed.

**Pull request merged**